packages/lodestar/src/chain/blocks/processor.ts

```diff
@@ -72,20 +74,85 @@ export async function processBlockJob(modules: BlockProcessorModules, job: IBloc
 /**
  * Similar to processBlockJob but this process a chain segment
  */
-export async function processChainSegmentJob(
-  modules: BlockProcessorModules,
-  chainSegmentJob: IChainSegmentJob
-): Promise<void> {
-  try {
-    const blockJobs: IBlockJob[] = chainSegmentJob.signedBlocks.map((signedBlock) => ({
-      signedBlock,
-      ...chainSegmentJob,
-    }));
-    await validateBlocks({...modules, jobs: blockJobs});
-    await processBlocks({...modules, jobs: blockJobs});
-  } catch (e) {
-    // above functions only throw BlockError
-    modules.emitter.emit(ChainEvent.errorBlock, e);
-    throw e;
-  }
+export async function processChainSegmentJob(modules: BlockProcessorModules, job: IChainSegmentJob): Promise<void> {
+  const blocks = job.signedBlocks;
+
+  // Validate and filter out irrelevant blocks
+  const filteredChainSegment: SignedBeaconBlock[] = [];
+  blocks.forEach(async (block, i) => {
```

Comment:
Would recommend to use
```ts
for (const [index, value] of arr.entries()) {
  //
}
```
instead

Reply:
Also `validateBlock` should not be async, thus this function either, see https://github.com/ChainSafe/lodestar/pull/1966/files#r559793686